### DISCUSSION TITLE
Adding the double harmonic major scale to ScalePattern

### DISCUSSION THREAD

**Discussion by JulioJamon54** [albertms10/music_notes#481] [Ideas]
Hi!
I hope this is the right place to discuss this with you, I'm not really use to this part of GitHub!
As I was browsing your module, I saw that you had the Octatonic scale but not the Double Harmonic scale, which surprised me.
So, could you add the double harmonic major scale to the ScalePattern? Or let me know if there is an easy way of using the available tools to make it myself (I saw that there is a fromBinary method, but I guess that I'll have to define its name manually etc).
Thank you! :)

**Comment by albertms10** [comment]
Hi Julio!

You’ve come to the right place for this discussion, and I appreciate your feedback! Regarding the Double harmonic scale, you’re right that it’s not included as a predefined static constant in the `ScalePattern` class. The constants provided are just a few examples since countless scales exist out there.

I’ll note this and add it to the next release, but it may take some time since I have a few busy weeks ahead.

The good news is that, in the meantime, you can define any scale using the `ScalePattern(List<Interval>)` constructor. For example, you can easily create the [Double harmonic scale](https://en.wikipedia.org/wiki/Double_harmonic_scale) like this:

```dart
const doubleHarmonicScale = ScalePattern([
  Interval.m2,
  Interval.A2,
  Interval.m2,
  Interval.M2,
  Interval.m2,
  Interval.A2,
  Interval.m2,
]);

doubleHarmonicScale.on(Note.c); // C null (C D♭ E F G A♭ B C)
```

This way, you can define and use the Double harmonic scale directly in your projects. I’m afraid the `ScalePattern.name` is still not quite extensible, but I plan to revisit it with extension in mind.

Nevertheless, if you believe any other scale could benefit other library users, I encourage you to contribute to the project by opening a new pull request with your additions.

Feel free to ask if you need any help with this process or have any more questions. Happy coding! 🧑‍💻

**Comment by JulioJamon54** [comment]
Hi Albert !

Thanks a lot for your very complete answer!
Don't worry if you take some time to release the scale officially, the solution you gave is really sufficient for my needs! :)
I might open a pull request if I find the time in the near future !

Thanks again for your kindness and the time you took to answer me! Good day ^^

**Comment by albertms10** [comment]
Addressed in #482. 🚀
